qemu/tests/pci_bridge_hotplug_unplug.py

```diff
@@ -0,0 +1,142 @@
+
+    devices += vm.devices.images_define_by_params(image, image_params, 'disk',
+                                                  None, False, None, pci_bus={'aobject': parent_bus})
+
+    for device in devices:
+        vm.devices.insert(device)
+        device.hotplug(vm.monitor)
+
+    ver_out = devices[-1].verify_hotplug("", vm.monitor)
+    if not ver_out:
+        test.fail("%s disk is not in qtree after hotplug!" % drive_format)
+
+    if drive_format == 'virtio':
+        return [devices[-1]]
+    else:
+        return devices[::2]
+
+
+def check_all_hotplugged_disks(test, params, env, image_num):
+    """
+    Check all hotplugged disks.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    error_context.context("Check 'info block'!", logging.info)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+
+    monitor_info_block = vm.monitor.info_block(False)
+    if image_num != len(monitor_info_block.keys()):
+        test.fail("The amount of block devices in 'info block' is incorrect!")
+    logging.info("Checking 'info block' pass!")
+
+    error_context.context("Read and write data on all hotplugged disks!", logging.info)
+    sub_test_type = params.get("sub_test_type", "dd_test")
+    images = params.get("images")
+    images = images.split()
+    images.pop(0)
+    for image in images:
+        params["dd_if"] = "ZERO"
+        params["dd_of"] = image
+        utils_test.run_virt_sub_test(test, params, env, sub_test_type)
+
+        params["dd_if"] = image
+        params["dd_of"] = "NULL"
+        utils_test.run_virt_sub_test(test, params, env, sub_test_type)
+    logging.info("Read and write data on all hotplugged disks successfully!")
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    [pci-bridge] Hotplug and unplug different devices to 1 pci-bridge
+    [pci-bridge] Hotplug and unplug different devices to different pci-bridges
+    This case will:
+    1) Start the guest.
+    2) Hotplug different devices to pci-bridge(s).
+    3) Check devices.
+    4) Hot-unplug devices.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    pci_bridge_num = int(params.get("pci_bridge_num", 0))
+    if pci_bridge_num > 1:
+        raise NotImplementedError()
+
+    error_context.context("Get the main VM", logging.info)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+
+    login_timeout = int(params.get("login_timeout", 360))
+    session = vm.wait_for_login(timeout=login_timeout)
+
+    image_list = params.get("images").split()
+    image_num = len(image_list)
+    if (image_num < 2):
+        test.fail("No pre-prepared images for hotplugging!")
+
+    device_list = []
+    fmt_list = params["disk_driver"].split()
+    if pci_bridge_num < 1:
```

Comment:
Line 98-99 are reserved for multi pci-bridges pre-process, and line 115, 117, 131 are specified for 0, 1, multi pci-bridge(s) cases. I think it's more clear. Thanks!